### PR TITLE
Split C++'s smoke_test into 4 stand alone ones

### PR DESCRIPTION
**Pull request by dgquintas**
namely:

test/cpp/qps/async_unary_ping_pong_test.cc
test/cpp/qps/qps_test.cc
test/cpp/qps/sync_streaming_ping_pong_test.cc
test/cpp/qps/sync_unary_ping_pong_test.cc

build.json contains the build rules for the newly introduced files. Makefile has been generated to reflect these changes.

### CODE REVIEW

**Comment by ctiller** [comment]
Lgtm once Travis is green

**Comment by nathanielmanistaatgoogle** [comment]
There are five commits in this pull request and four of them appear to have no relevance to the content of this change as described in the pull request itself. May I trouble you to clean up the history drafted here, and to describe the Makefile changes in the pull request description?

**Comment by dgquintas** [comment]
@nathanielmanistaatgoogle absolutely, apologies, my github-fu is weak.

I suspect this is due to having based this branch on another one that's still pending. In general, I wonder how to go about cleaning up the commit history. 

I've added more details to the description.

**Comment by ctiller** [comment]
Can you merge this with head and re-push?

**Comment by dgquintas** [comment]
@ctiller Done.
@nathanielmanistaatgoogle in trying to clean the history (hadn't done it before), I actually made it worse. I'll get it right next time, especially before pushing. Sorry about that.

File: build.json [hunk 1]
```diff
@@ -1838,6 +1838,24 @@
       ]
     },
     {
+      "name": "async_unary_ping_pong_test",
```
Comment (ctiller) [inline build.json]: I think you also need to delete qps_smoke_test from here?

**Comment by nathanielmanistaatgoogle** [comment]
@dgquintas it's not quite clear to me what you're saying. Do you feel that the commits drafted in this as-yet-unmerged pull request should be as-they-are-now incorporated into canonical project history?

**Comment by dgquintas** [comment]
@nathanielmanistaatgoogle Victory! Thanks for bringing up the issue. Git is hard. I'm still not quite sure how I did it...

**Comment by nathanielmanistaatgoogle** [comment]
I appreciate your diligence; thank you.

**Comment by ctiller** [comment]
LGTM. Looks like we've got a merge conflict however.
Really looking forward to merging this!

**Pull request merged**